### PR TITLE
XML documentation fixes

### PR DESCRIPTION
**Pull request by amoerie**
DocFx produces a whole list of warnings when it runs.
This PR fixes those warnings.
There are no actual code changes, only changes in the XML annotations of public types, properties or methods.

#### Checklist
- [x] The pull request branch is in sync with latest commit on the *fo-dicom/development* branch
- [ ] I have updated API documentation (not needed)
- [ ] I have included unit tests (not needed)
- [ ] I have updated the change log (not needed)
- [x] I am listed in the CONTRIBUTORS file

#### Changes proposed in this pull request:
- Fix a bunch of incorrect XML documentation annotations

### CODE REVIEW

File: FO-DICOM.Core/Network/DicomCStoreRequest.cs [hunk 1]
```diff
@@ -9,30 +9,6 @@ namespace FellowOakDicom.Network
     /// <summary>
     /// Represents a DICOM C-Store request to be sent to a C-Store SCP or a C-Store request that has been received from a C-Store SCU.
     /// </summary>
-    /// <example>
-    /// The following example shows how to use the <see cref="DicomClient"/> class to send DICOM C-Store requests to a DICOM C-Store SCP.
```
Comment (amoerie) [inline FO-DICOM.Core/Network/DicomCStoreRequest.cs:13]: This sample was completely out of date, and imho not in the right place. We have plenty of examples in the README and in unit tests.

**Comment by amoerie** [comment]
Amazingly, the build fails because of a flaky test.
I've created a separate PR for that here: https://github.com/fo-dicom/fo-dicom/pull/1554

File: FO-DICOM.Core/DicomTransferSyntax.cs [hunk 2]
```diff
@@ -77,10 +77,8 @@ private DicomTransferSyntax(DicomUID uid)
 
         #region METHODS
 
-        /// <inheritdoc />
```
Comment (mrbean-bremen) [inline FO-DICOM.Core/DicomTransferSyntax.cs:80]: Is this not needed anymore?

File: FO-DICOM.Core/Imaging/BitDepth.cs [hunk 3]
```diff
@@ -112,10 +112,10 @@ public static int GetHighBit(int bitsStored, bool isSigned)
         }
 
         /// <summary>
-        /// Create new instance of <seealso cref="BitDepth"/> from input <paramref name="dataset"/>
```
Comment (mrbean-bremen) [inline FO-DICOM.Core/Imaging/BitDepth.cs:115]: Do these generate warnings? They look ok to me (apart from the fact that they probably create the same output as `see`).

File: FO-DICOM.Core/DicomTransferSyntax.cs (hunk 2 above)
Comment (amoerie) [reply FO-DICOM.Core/DicomTransferSyntax.cs:80]: No these methods are documented in System.Object and there is no need to explicitly inherit those docs. In fact, we already don't in other places such as DicomTag, which also has an Equals and GetHashCode implementation.

File: FO-DICOM.Core/Imaging/BitDepth.cs (hunk 3 above)
Comment (amoerie) [reply FO-DICOM.Core/Imaging/BitDepth.cs:115]: Hmmm this looks like a mistake! 
Edit: no it isn't, Github mobile was misleading.

File: FO-DICOM.Core/DicomTransferSyntax.cs (hunk 2 above)
Comment (mrbean-bremen) [reply FO-DICOM.Core/DicomTransferSyntax.cs:80]: Ah ok, makes sense.

File: FO-DICOM.Core/Imaging/BitDepth.cs (hunk 3 above)
Comment (amoerie) [reply FO-DICOM.Core/Imaging/BitDepth.cs:115]: @mrbean-bremen you should never put a "seealso" tag inside a "summary" tag.
See the official docs here: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/recommended-tags#seealso 

Quoting from that link (emphasis mine):

> The <seealso> tag lets you specify the text that you might want to appear in a See Also section. Use [<see>](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/recommended-tags#see) to specify a link from within text. **You cannot nest the seealso tag inside the summary tag**.

Conceptually, the "see" tag functions much like how an "a" tag with a href would in HTML.
The "seealso" tag can be compared with a footer with extra links.

Comment (amoerie) [reply FO-DICOM.Core/Imaging/BitDepth.cs:115]: P.S. I distinctly remember getting warnings about this when I made the original commit (it was cherry picked here) but I cannot seem to reproduce them today. Still, as per my previous comment, it's still better to keep the changes imho.

**Review by mrbean-bremen — APPROVED** [review]

**Review by gofal — APPROVED** [review]

**Pull request merged**